### PR TITLE
[Tests]Use Eventually in test to allow time for reconciliation loop

### PR DESCRIPTION
**Pull request by Barakmor1**
Use Eventually in test to allow time for reconciliation loop to update VM after VMI removal.

fix flake:
https://prow.ci.kubevirt.io/view/gcs/kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.30-sig-compute/1837630591583916032

### What this PR does
Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/vm_test.go [hunk 1]
```diff
@@ -582,14 +582,15 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
...
-			Expect(err).ToNot(HaveOccurred())
+			By("Ensuring stateChangeRequests list gets cleared")
+			Eventually(func() []v1.VirtualMachineStateChangeRequest {
+				vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return vm.Status.StateChangeRequests
+			}, 60*time.Second, 1*time.Second).Should(BeEmpty(), "VirtualMachine should reflect VMI deletion")
+			Expect(vm.Spec.RunStrategy).ToNot(BeNil())
 			Expect(vm.Spec.RunStrategy).ToNot(BeNil())
 			Expect(*vm.Spec.RunStrategy).To(Equal(expectedRunStrategy))
```
Comment (fossedihelm) [inline tests/vm_test.go:593]:
```golang
			By("Ensuring stateChangeRequests list gets cleared")
			// StateChangeRequest might still exist until the new VMI is created
			// But it must eventually be cleared
			Eventually(ThisVM(vm), 240*time.Second, 1*time.Second).Should(Not(HaveStateChangeRequests()))
			Expect(vm.Spec.RunStrategy).ToNot(BeNil())
			Expect(*vm.Spec.RunStrategy).To(Equal(expectedRunStrategy))
```
 :slightly_smiling_face:

Comment (Barakmor1) [reply tests/vm_test.go]: Done

**Review by fossedihelm — APPROVED** [review]
Thank you!

File: tests/vm_test.go [hunk 2]
```diff
@@ -582,14 +582,10 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 
 			By("Ensuring the VirtualMachineInstance is removed")
 			Eventually(ThisVMIWith(vm.Namespace, vm.Name), 240*time.Second, 1*time.Second).ShouldNot(Exist())
-
-			vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
-			Expect(err).ToNot(HaveOccurred())
+			By("Ensuring stateChangeRequests list gets cleared")
+			Eventually(ThisVM(vm), 60*time.Second, 1*time.Second).Should(Not(HaveStateChangeRequests()))
```
Comment (xpivarc) [inline tests/vm_test.go:586]: The below, assert would fail because we do not fetch the vm again. Also the timeout should be lower.

Comment (Barakmor1) [reply tests/vm_test.go:586]: Does 30 seconds works ?

**Pull request merged**